confluent_kafka/schema_registry/json_schema.py

```diff
@@ -0,0 +1,261 @@
+from io import BytesIO
+
+import json
+import struct
+
+from jsonschema import validate, ValidationError
+
+from confluent_kafka.schema_registry import (_MAGIC_BYTE,
+                                             Schema,
+                                             topic_subject_name_strategy)
+from confluent_kafka.serialization import SerializationError, Serializer, Deserializer
+
+
+class _ContextStringIO(BytesIO):
+    """
+    Wrapper to allow use of StringIO via 'with' constructs.
+
+    """
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *args):
+        self.close()
+        return False
+
+
+class JsonSerializer(Serializer):
+    """
+    JsonSerializer serializes objects in the Confluent Schema Registry binary
+    format for JSON.
+
+    Note:
+        The ``title`` annotation, referred to as a record name
+        elsewhere in this document, is not strictly required by the JSON Schema
+        specification. It is however required by this Serializer. This
+        annotation(record name) is used to register the Schema with the Schema
+        Registry. See documentation below  for additional details on Subjects
+        and schema registration.
+
+    JsonSerializer configuration properties:
+    +-----------------------+----------+--------------------------------------------------+
+    | Property Name         | type     | Description                                      |
+    +=======================+==========+==================================================+
+    |                       |          | Registers schemas automatically if not           |
+    | auto.register.schemas | bool     | previously associated with a particular subject. |
+    |                       |          | Defaults to True.                                |
+    +-----------------------|----------+--------------------------------------------------+
+    |                       |          | Callable(str, SerialalizationContext) -> str     |
```

Comment:
the arg order in this comment doesn't match reality anymore. you need to update the avro doc too, and i'm guessing protobuf.